### PR TITLE
Migrate provided id test

### PR DESCRIPTION
**Pull request by v9n**
We have updated our chain API in runtime 2.9.5 and oak-collator 2.0.0. it is a breaking changes so we updated this to reflect our task id change.

Before:

task id can be infer on client side using provided id

After:

client need to fetch events and find task id to assert or do more CRUD.

Test result run againt a local node on `bump-to-2.0.0` branch on revision `e3b6896bf6d6d9f07a4fdbe70155b8cccfe140f1`

```
 PASS  test/functional/rpc.test.ts
  ✓ scheduler.getTimeAutomationFees works (144 ms)
  ✓ scheduler.calculateOptimalAutostaking works (104 ms)
  ✓ scheduler.getAutoCompoundDelegatedStakeTaskIds works (95 ms)

Test Suites: 3 passed, 3 total
Tests:       47 passed, 47 total
Snapshots:   0 total
Time:        682.918 s, estimated 7029 s
```

### CODE REVIEW

File: test/functional/cancel.test.ts [hunk 1]
```diff
@@ -20,14 +20,6 @@ const initialize = async () => {
 beforeEach(() => initialize());
 afterEach(() => polkadotApi.disconnect());
 
-test('Cancel failed with incorrect format taskID', async () => {
```
Comment (chrisli30) [inline test/functional/cancel.test.ts:23]: The **incorrect format** taskId should still be tested during cancellation. Is it covered by other tests already?

File: test/utils/helpFn.ts [hunk 2]
```diff
@@ -71,7 +66,7 @@ export const getContext = async (polkadotApi: ApiPromise) => {
  
            const event = _.find(events, ({ event }) => polkadotApi.events.system.ExtrinsicSuccess.is(event));
            if (event) {
-             resolve({ extrinsicHash: txHash, blockHash: status?.asFinalized?.toString() });
+             resolve({ extrinsicHash: txHash, blockHash: status?.asFinalized?.toString(), events });
```
Comment (chrisli30) [inline test/utils/helpFn.ts:69]: Do you want to pass in all `events` or a selected `event` at this line? The `if(event)` condition doesn’t look necessary to me.

**Review by chrisli30 — COMMENTED** [review]
Looks good to me overall, but I have got two comments.

File: test/utils/helpFn.ts (hunk 2 above)
Comment (imstar15) [reply test/utils/helpFn.ts:69]: In this context, "events" is a **return** value that includes all the events generated by the extrinsic.

I think the `if(event)` condition is necessary. It checks whether the extrinsic execution was successful.

**Review by imstar15 — CHANGES_REQUESTED** [review]
In this file, there are still some places that need to be modified.

https://github.com/OAK-Foundation/oak.js/blob/f5ef74c9246fc45a2fb265be528146a81bfa020c/packages/types/src/automationTime.ts#L3

File: test/functional/cancel.test.ts (hunk 1 above)
Comment (chrisli30) [reply test/functional/cancel.test.ts:23]: Right, with the new taskId "a string here" won’t be able to find anything.

File: test/functional/cancel.test.ts [hunk 3]
```diff
@@ -20,14 +20,6 @@ const initialize = async () => {
...
-  const nonexistentTaskID = "A string(<32 bytes).";
-  await checkBalance(polkadotApi, keyringPair);
-
-  // Cancel failed with incorrect format taskID
-  await expect(automationTimeApi.buildCancelTaskExtrinsic(keyringPair, nonexistentTaskID)).rejects.toBeInstanceOf(Error);
-});
-
 test('Cancel failed with nonexistent taskID', async () => {
```
[the remark this answers is not in the document]
Comment (chrisli30) [reply test/functional/cancel.test.ts:23]: yes, that makes sense since all taskId are generated by the chain and follow a strict format now.

**Review by chrisli30 — DISMISSED** [review]

File: test/utils/helpFn.ts (hunk 2 above)
Comment (chrisli30) [reply test/utils/helpFn.ts:69]: Okay, got it. Could you add these comments to the line above the `if(event)` to explain it?

File: test/utils/helpFn.ts [hunk 4]
```diff
@@ -154,7 +149,7 @@ export const cancelTaskAndVerify = async (automationTimeApi: AutomationTimeApi,
 
   expect(section).toEqual(SECTION_NAME);
   expect(method).toEqual('cancelTask');
-  expect(taskIdOnChain.toString()).toEqual(taskID);
+  expect(new Buffer(taskIdOnChain).toString()).toEqual(taskID);
```
Comment (imstar15) [inline test/utils/helpFn.ts:152]: Why is `new Buffer(...)` being used here?

**Review by imstar15 — APPROVED** [review]
Looks good to me!

**Review by imstar15 — CHANGES_REQUESTED** [review]
`/packages/types/src/xcmpHandler.ts` is also needed to change.


https://github.com/OAK-Foundation/oak.js/blob/f5ef74c9246fc45a2fb265be528146a81bfa020c/packages/types/src/xcmpHandler.ts#L23

**Review by chrisli30 — DISMISSED** [review]
It’s good to go!

**Review by imstar15 — APPROVED** [review]
Looks good to me!

**Pull request merged**